core/field_label.js

```diff
@@ -36,15 +36,16 @@ goog.require('goog.math.Size');
 
 /**
  * Class for a non-editable, non-serializable text field.
- * @param {string} text The initial content of the field.
+ * @param {string=} text The initial content of the field, defaults to an
+ *    empty string.
  * @param {string=} opt_class Optional CSS class for the field's text.
  * @extends {Blockly.Field}
  * @constructor
  */
 Blockly.FieldLabel = function(text, opt_class) {
   this.size_ = new goog.math.Size(0, 17.5);
   this.class_ = opt_class;
-  this.setValue(text);
+  this.setValue(text || '');
```

Comment:
I didn't think enough about this before. In your tests you require 1 -> '1' and true -> 'true', but not 0 -> '0' and false -> 'false' . It looks like those values will result in the empty string instead.

Here and elsewhere can we update this to explicitly check for null/undefined instead and let the majority of things be auto-cast to strings?

Reply:
Fixed for all applicable fields.